cmd/osbuild-composer/composer.go

```diff
@@ -78,16 +78,29 @@ func NewComposer(config *ComposerConfigFile, stateDir, cacheDir string) (*Compos
 	}
 
 	repoConfigs, err := reporegistry.LoadAllRepositories(repositoryConfigs)
-	if err != nil {
-		return nil, fmt.Errorf("error loading repository definitions: %v", err)
+	switch err.(type) {
+	case *reporegistry.NoReposLoadedError:
+		// When running as a service (cloud API), there are no repositories
+		// configured, so this is OK. The weldr initialiser should check if
+		// there are no repositories configured and behave accordingly.
+		// Log for troubleshooting.
+		logrus.Info(err.Error())
```

Comment:
Although it is not a great solution, I would be more in favor for adding a top level osbuild-composer configuration option, such as `ignore_missing_repos`, and failing if it is not set and there were no repos loaded.

We can then set the option by default in the [composer OSP template](https://github.com/osbuild/osbuild-composer/blob/main/templates/openshift/composer.yml).

The reason is that even [Cloud API may use the loaded repositories](https://github.com/osbuild/osbuild-composer/blob/main/internal/cloudapi/v2/compose.go#L1161-L1169). With the server-less on-prem effort, I'm not sure if we will end up replacing Weldr API with Cloud API on-prem, before removing the server part completely.

But I'm happy to be convinced otherwise :innocent:

Reply:
ooooh, was not aware.  No, I will not convince you otherwise.

Let's do the config.

Reply:
Done!